xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -294,9 +294,7 @@ private void updateWeight() {
         double newWeight = subchannel.getWeight();
         newWeights[i] = newWeight > 0 ? (float) newWeight : 0.0f;
       }
-
-      StaticStrideScheduler scheduler = new StaticStrideScheduler(newWeights, random);
-      this.scheduler = scheduler;
+      this.scheduler = new StaticStrideScheduler(newWeights, new AtomicInteger(random.nextInt()));
```

Comment:
WRRLB should have a final AtomicInteger that can be initialized to 0. It then should pass that same instance to each of the schedulers. The point was to not reset the sequence, which is a difference in behavior.

Reply:
Got it. I understand passing the same scheduler throughout the weight updates, but don't we want to randomize the initial state?

Reply:
No, we don't need to randomize. The reason to randomize was to make sure we don't over-value the very first entry when regenerating the scheduler. If we always started at 0, and we frequently generated the scheduler, then the first entries could be sent more traffic.

But if we instead continue counting where we left off, then if we re-create the scheduler we _don't_ favor 0. We continue wherever we were, or some "random" place if the number of channels changed.

Reply:
Right. I'm following. But, what if there are multiple clients sending traffic? In this case, wouldn't initializing the sequence to 0 end up favoring the first entry? Or does the first entry point to different backends for different clients?

Reply:
Yes, you could skew with many very-short-lived clients. But I didn't think we cared about that.

The gRFC doesn't really say in a clear way. Looking at Go and C, they did randomize this number on creation. So we should do the same. On the plus side, that helps test negative number handling.